src/app/components/AnalyticsConsent/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react'
+import { useLocation } from 'react-router-dom'
+import Snackbar from '@mui/material/Snackbar'
+import Typography from '@mui/material/Typography'
+import { styled } from '@mui/material/styles'
+import Button from '@mui/material/Button'
+import Link from '@mui/material/Link'
+import { Trans, useTranslation } from 'react-i18next'
+import Card from '@mui/material/Card'
+import CardContent from '@mui/material/CardContent'
+import CardActions from '@mui/material/CardActions'
+import { useScreenSize } from 'app/hooks/useScreensize'
+import * as matomo from './initializeMatomo'
+import { legalDocuments } from '../../utils/externalLinks'
+
+matomo.loadMatomo()
+
+const AcceptCookiesButton = styled(Button)(({ theme }) => ({
+  paddingLeft: theme.spacing(5),
+  paddingRight: theme.spacing(5),
+}))
+
+const DeclineCookiesButton = styled(Button)(({ theme }) => ({
+  paddingLeft: theme.spacing(5),
+  paddingRight: theme.spacing(5),
+}))
+
+export const AnalyticsConsent = () => {
+  const { t } = useTranslation()
+  const { isMobile } = useScreenSize()
+  const [hasAccepted, setHasAccepted] = useState<matomo.HasAccepted>('timed_out_matomo_not_loaded')
+
+  useEffect(() => {
+    const updateHasAccepted = async () => {
+      setHasAccepted(await matomo.hasAccepted({ timeout: 10_000 }))
+    }
+    updateHasAccepted()
+  }, [])
```

Comment:
Due to how we implemented layout+footer, this gets called a lot.

Reply:
You could wrap this in a custom hook, and debounce it(singleton approach). Not sure how heavy `matomo.hasAccepted` is. One thing that is missing here is definitely clean up function(in case component gets destroyed in the meantime, `setHasAccepted` should not be called), if I understand correctly, you are saying that this gets called for every navigation - thats why it gets called a lot.

Reply:
Or does `matomo.hasAccepted({ timeout: 10_000 })`, `timeout` parameter mean that it has 10s debouce already? Then I don't see an issue.

Reply:
Issue is the whole component gets destroyed and recreated on every navigation

(I don't see any issue or need for a cleanup function)